domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs

```diff
@@ -531,4 +568,36 @@ public async Task CreateWalletDepositEndpoint_ShouldReturnUnauthorized_WhenUnaut
 
         result.StatusCode.Should().Be(HttpStatusCode.Unauthorized);
     }
+
+    [Fact]
+    public async Task Create_ShouldHaveReceiverReferenceNotNull()
+    {
+        var receiverId = Guid.NewGuid();
+
+        var result = await factory
+            .CreateAuthenticatedClient(sub: receiverId.ToString())
+            .PostAsync("api/transfer-agreements/wallet-deposit-endpoint", null);
+
+        var resultData = await result.Content.ReadFromJsonAsync<Dictionary<string, string>>();
+        var base64String = resultData!["result"];
+
+        var transferAgreement = new CreateTransferAgreement(
+            new DateTimeOffset(2123, 3, 3, 3, 3, 3, TimeSpan.Zero).ToUnixTimeSeconds(),
+            new DateTimeOffset(2124, 4, 4, 4, 4, 4, TimeSpan.Zero).ToUnixTimeSeconds(),
+            "12345456",
+            base64String
+        );
+
+        var response = await authenticatedClient.PostAsJsonAsync("api/transfer-agreements", transferAgreement);
+        response.EnsureSuccessStatusCode();
+
+        var createdTransferAgreement = await response.Content.ReadFromJsonAsync<TransferAgreementDto>();
+
+        await Task.Delay(2000);
```

Comment:
Why a delay here?

Reply:
Because the builder is too slow build and host the ports for the test containers, so we are giving it a 2 second delay, so it has time to build the container, before it starts searching for its host port.